### ISSUE TITLE
Provide a clearer error message when dotnet-install-tool fails because of incorrect package type

### ISSUE THREAD

**Issue by natemcmaster** [dotnet/cli#8698]
## Steps to reproduce
`dotnet install tool Microsoft.Dotnet.Watcher.Tools -g`

## Expected  behavior
A message that explains the package clearly what is wrong. Maybe:

> This package does not support installation as a global command line tool because it does not have the correct package type in its metadata. See https://aka.ms/something for details on requirements for command line tool packages.

## Actual behavior
You get a wall of text. The error message does not clearly indicate what went wrong.

```
> dotnet install tool Microsoft.Dotnet.Watcher.Tools -g
Install failed. Failed to download package:
NuGet returned:

Failed to restore package.
WorkingDirectory:
Arguments: restore C:\Users\scott\AppData\Local\Temp\gnzjrzwo.bvk\nh3id1v4.onz.csproj --runtime win10-x64 /p:BaseIntermediateOutputPath=\"C:\Users\scott\.dotnet\toolspkgs\Microsoft.Dotnet.Watcher.Tools\2gtgbll5.42m\"
Output:   Restoring packages for C:\Users\scott\AppData\Local\Temp\gnzjrzwo.bvk\nh3id1v4.onz.csproj...
  Installing Microsoft.NETCore.DotNetAppHost 2.0.0.
  Installing Microsoft.NETCore.DotNetHostResolver 2.0.0.
  Installing NETStandard.Library 2.0.0.
  Installing Microsoft.NETCore.Platforms 2.0.0.
  Installing Microsoft.NETCore.DotNetHostPolicy 2.0.0.
  Installing Microsoft.NETCore.App 2.0.0.
  Installing Microsoft.DotNet.Watcher.Tools 2.0.0.
  Installing runtime.win-x64.Microsoft.NETCore.DotNetAppHost 2.0.0.
  Installing runtime.win-x64.Microsoft.NETCore.DotNetHostResolver 2.0.0.
  Installing runtime.win-x64.Microsoft.NETCore.DotNetHostPolicy 2.0.0.
  Installing runtime.win-x64.Microsoft.NETCore.App 2.0.0.
C:\Users\scott\AppData\Local\Temp\gnzjrzwo.bvk\nh3id1v4.onz.csproj : error NU1212: Invalid project-package combination for Microsoft.DotNet.Watcher.Tools 2.0.0. DotnetToolReference project style can only contain references of the DotnetTool type
  Restore failed in 5.89 sec for C:\Users\scott\AppData\Local\Temp\gnzjrzwo.bvk\nh3id1v4.onz.csproj.

```

The error message is

> Invalid project-package combination for Microsoft.DotNet.Watcher.Tools 2.0.0. DotnetToolReference project style can only contain references of the DotnetTool type

I don't think this error makes it clear to users why the package could not be installed.

## Environment data
`dotnet --version`: 2.1.300-preview1-008174

cc @wli3 @KathleenDollard @shanselman 

cref https://github.com/aspnet/DotNetTools/issues/399 and https://github.com/aspnet/Announcements/issues/290#issuecomment-369015372

**Comment by wli3** [comment]
One of the decision is let nuget error pass through and cli will not do further "translation". @KathleenDollard @peterhuene @livarcocc in this case we should ask nuget to have better error message do you agree?

**Comment by KathleenDollard** [comment]
We had some internal discussion of this, and decided that failing to pass the NuGet message through crippled certain debugging scenarios. Specifically, if a package can't be found, knowing which package is to found is helpful. 

Yes, it is painful here and it is going to be an easy mistake to make.  

Let's take another quick look at this

**Comment by shanselman** [comment]
Thanks! I feel that dotnet.exe is quickly becoming (and should be considered) a first line "User Interface" so I'd love to see less logging and stack traces and more "pit of success" messages.

**Comment by peterhuene** [comment]
For reference, here's what's currently displayed for preview2 (master):

```
$ dotnet install tool Microsoft.Dotnet.Watcher.Tools -g
/var/folders/rl/m1rv1rxd643gpwspqrbxy76r0000gn/T/w1heen5l.wc5/yh505uu5.u4i.csproj : error NU1212: Invalid project-package combination for Microsoft.DotNet.Watcher.Tools 2.0.0. DotnetToolReference project style can only contain references of the DotnetTool type
The tool package could not be restored.
Tool 'Microsoft.Dotnet.Watcher.Tools' failed to install.
```

**Comment by peterhuene** [comment]
I've filed #8707 to at least remove the temp project from the error message since it is entirely unhelpful and only contributes to a worse diagnostic experience.  We'll file a issue with NuGet to see if we can improve the underlying error message when package type isn't a "tool".

**Comment by KathleenDollard** [comment]
I have filed https://github.com/NuGet/Home/issues/6630 to work on the text. Without that fix from NuGet, the experience will now be 

```
$ dotnet install tool Microsoft.Dotnet.Watcher.Tools -g
error NU1212: Invalid project-package combination for Microsoft.DotNet.Watcher.Tools 2.0.0. DotnetToolReference project style can only contain references of the DotnetTool type
The tool package could not be restored.
Tool 'Microsoft.Dotnet.Watcher.Tools' failed to install.
```

With my suggestion for the NuGet fix (open for better language, please comment in the NuGet issue)

```
$ dotnet install tool Microsoft.Dotnet.Watcher.Tools -g
The requested package does not have the expected DotnetTool type and can't be used where that type is expected. For example, this package can't be used with "dotnet install tool -g".
The tool package could not be restored.
Tool 'Microsoft.Dotnet.Watcher.Tools' failed to install.
```

@natemcmaster and others: Do you think this solves this problem?

**Comment by natemcmaster** [comment]
Yeah, I think that would work.

**Comment by peterhuene** [comment]
@livarcocc Do we need this held open? The NuGet issue Kathleen opened is where the real fix needs to happen and #8707 has already been fixed.

**Comment by peterhuene** [comment]
Now that the other clean-up work is in for the tool install UX, I'm going to close this issue in favor of Nuget/Home#6630 that is the remaining piece of the puzzle (a clearer message that the package that was found was not a tool package and cannot be installed with `dotnet tool install`).
